spec/bundler/stub_specification_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+if Bundler.rubygems.provides?(">= 2.1")
+  RSpec.describe Bundler::StubSpecification do
+    let(:with_gem_stub_spec) do
+      stub = Gem::Specification.stubs.first
+      described_class.from_stub(stub)
+    end
+
+    let(:with_bundler_stub_spec) do
```

Comment:
The issue is that in some cases `stub` ends up being `Bundler::StubSpecification`. When someone loads a gem with `Gem::Specification.find_by_name(*args)` after `require 'bundler/setup'`, it loads it again with a cache (it seems).